pymc_marketing/mmm/base.py

```diff
@@ -54,13 +55,19 @@ def __init__(
         sampler_config: Optional[Dict] = None,
         **kwargs,
     ) -> None:
-        self.X: Optional[pd.DataFrame] = None
-        self.y: Optional[Union[pd.Series, np.ndarray]] = None
         self.date_column: str = date_column
         self.channel_columns: Union[List[str], Tuple[str]] = channel_columns
+
         self.n_channel: int = len(channel_columns)
-        self._fit_result: Optional[az.InferenceData] = None
-        self._posterior_predictive: Optional[az.InferenceData] = None
+
+        self.X: Optional[pd.DataFrame] = None
+        self.y: Optional[Union[pd.Series, np.ndarray]] = None
+
+        self._time_resolution: int
+        self._time_index: NDArray[np.int_]
+        self._time_index_mid: int | float
```

Comment:
I think that this would be nice to package up together. These three items are related and can get easily lost in the large class. 

It'd be a relatively small class but would lump some of these specific calculation together

Some pseudo code: 
```python
class TimeIndex: 
    def __init__(self) -> None: 
        self._time_resolution: int
        self._time_index: NDArray[np.int_]
        self._time_index_mid: int | float

    def initialize_values(X: pd.DataFrame, date_column: str) -> None:
        self._time_index = np.arange(0, X.shape[0])
        self._time_index_mid = X.shape[0] // 2
        self._time_resolution = (
            X[date_column].iloc[1] - X[date_column].iloc[0]
        ).days

    @property 
    def defaults(self): 
        L = (
            self._time_index_mid + DAYS_IN_YEAR / self._time_resolution
        )
        # more related calculation
        
    # More related methods
```

A single class instance could be an attribute of the MMM class then rather 3 different things that keep needing "used by TVP" etc

What are your thoughts?

Reply:
I like it :)

Reply:
Very nice. Will resolve all the little things, then revisit this.

Reply:
Played around with an implementation of this, and will leave it for now. I think it makes more sense to add this in a later PR or as part of a major refactor.